tanjun/annotations.py

```diff
@@ -140,16 +140,39 @@ def set_config(self, config: _ArgConfig, /) -> None:
 """An argument which takes a bool-like value."""
 
 Channel = typing.Annotated[hikari.PartialChannel, _OptionMarker(hikari.PartialChannel)]
-"""An argument which takes a channel."""
+"""An argument which takes a channel.
+
+[hikari.InteractionChannel][hikari.interactions.command_interactions.InteractionChannel]
+will be passed for options typed as this when being called as a slash command.
+"""
+
+InteractionChannel = typing.Annotated[hikari.InteractionChannel, _OptionMarker(hikari.InteractionChannel)]
+"""An argument which takes a channel with interaction specific metadata.
+
+This will not work for message commands (unlike
+[annotations.Channel][tanjun.annotations.Channel]).
+"""
 
 Float = typing.Annotated[float, _OptionMarker(float)]
 """An argument which takes a floating point number."""
 
 Int = typing.Annotated[int, _OptionMarker(int)]
 """An argument which takes an integer."""
 
+
 Member = typing.Annotated[hikari.Member, _OptionMarker(hikari.Member)]
-"""An argument which takes a guild member."""
+"""An argument which takes a guild member.
+
+[hikari.InteractionMember][hikari.interactions.base_interactions.InteractionMember]
+will be passed for options typed as this when being called as a slash command.
+"""
+
+InteractionMember = typing.Annotated[hikari.InteractionMember, _OptionMarker(hikari.InteractionMember)]
```

Comment:
As these types to the decorator function's docs with a note that they're app command exclusive